### ISSUE TITLE
pipe do not emit finish in v0.10.30

### ISSUE THREAD

**Issue by dead-horse** [nodejs/node-v0.x-archive#8186]
the whole test code is in: https://gist.github.com/dead-horse/c7030afa663fee61028b

if response connection is not keepalive: 

``` js
var http = require('http');
var fs = require('fs');
var path = require('path');
var tmp = path.join(__dirname, 't.tgz');

var opts = {
  hostname: 'registry.npmjs.com',
  method: 'GET',
  port: 80,
  path: '/cutter/-/cutter-0.0.3.tgz',
  headers: {
    Connection: 'close'
  }
}

http.request(opts, function (res) {
  console.log('connection: %s', res.headers.connection);
  res.resume();
  res.pause();
  setTimeout(function () {
    var tarball = fs.createWriteStream(tmp, { mode: 420 })
    tarball.on('finish', function () {
      console.log('tarball finish');
    })
    res.on('data', function (data) {
      console.log('res on data: %s', data.length);
    })
    res.on('end', function (d) {
      console.log('res end');
    })
    res.pipe(tarball);
    res.resume();
  }, 1000);

}).end();

// output:
// connection: close
// res on data: 3144
// res end
```

`pipe` won't emit `finish` event in tarball.

see https://github.com/npm/npm/pull/5958

**Comment by jasnell** [comment]
@chrisdickinson ... was there any resolution on this?

**Comment by jasnell** [comment]
This appears to have been fixed in v0.12. Possibly still an issue in v0.10.
